aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -387,15 +389,24 @@ module aptos_framework::multisig_account {
     }
 
     #[view]
-    /// Return the payload for the next transaction in the queue.
+    /// Return the payload for the next transaction in the queue. If no payload is stored on chain, the provided
+    /// payload is returned. If a payload is stored on chain and a non-empty payload is provided, the provided payload
+    /// must match the on-chain payload. Otherwise, this function will abort.
     public fun get_next_transaction_payload(
         multisig_account: address, provided_payload: vector<u8>): vector<u8> acquires MultisigAccount {
         let multisig_account_resource = borrow_global<MultisigAccount>(multisig_account);
         let sequence_number = multisig_account_resource.last_executed_sequence_number + 1;
         let transaction = table::borrow(&multisig_account_resource.transactions, sequence_number);
 
         if (option::is_some(&transaction.payload)) {
-            *option::borrow(&transaction.payload)
+            let onchain_payload = option::borrow(&transaction.payload);
+            if(features::multisig_v2_fix_enabled()) {
+                assert!(
```

Comment:
nit: Can be combined with the previous if statement: assert!(!features::multisig_v2_fix_enabled() || ... )